packages/astro/src/core/preview/vite-plugin-astro-preview.ts

```diff
@@ -61,7 +63,19 @@ export function vitePluginAstroPreview(settings: AstroSettings): Plugin {
 						res.statusCode = 404;
 						res.end(notFoundTemplate(pathname, 'Not Found'));
 					}
-				});
+				};
+
+				// Vite 5 has its own 404 middleware, we replace it with ours instead.
+				if (IS_VITE_5) {
+					for (const middleware of server.middlewares.stack) {
+						// This hardcoded name will not break between Vite versions
+						if ((middleware.handle as any).name === 'vite404Middleware') {
```

Comment:
```suggestion
						if ((middleware.handle as any)?.name === 'vite404Middleware') {
```
Is `.handle.name` safe with `any`? Should we have the optional chain operator?

Reply:
It should always be defined when a middleware is added. Actually it seems like I can use `as Connect.HandleFunction` instead. Will switch to that so it's more explicit.